src/NServiceBus.Metrics/MetricsOptions.cs

```diff
@@ -19,69 +11,131 @@ public class MetricsOptions
         /// Enables sending periodic updates of metric data to ServiceControl
         /// </summary>
         /// <param name="serviceControlMetricsAddress">The transport address of the ServiceControl instance</param>
-        /// <param name="interval">How frequently metric data is sent to ServiceControl</param>
+        /// <param name="interval">Inteval between consequitive reports</param>
         [Obsolete("Not for public use.")]
         public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval)
         {
             Guard.AgainstNullAndEmpty(nameof(serviceControlMetricsAddress), serviceControlMetricsAddress);
             Guard.AgainstNegativeAndZero(nameof(interval), interval);
 
-            reportInstallers.Add((builder, config) => config.WithReport(
-                new NServiceBusMetricReport(builder.Build<IDispatchMessages>(), serviceControlMetricsAddress, builder.Build<HostInformation>()),
-                interval
-            ));
+            ServiceControlMetricsAddress = serviceControlMetricsAddress;
+            ReportingInterval = interval;
         }
 
         /// <summary>
-        /// Enables sending metric data to the trace log
+        /// Enables registering observers to available probes.
         /// </summary>
-        /// <param name="interval">How often metric data is sent to the trace log</param>
-        public void EnableMetricTracing(TimeSpan interval)
+        /// <param name="register">Action that registers observers to probes</param>
+        public void RegisterObservers(Action<ProbeContext> register)
         {
-            Guard.AgainstNegativeAndZero(nameof(interval), interval);
+            Guard.AgainstNull(nameof(register), register);
+
+            registerObservers += register;
+        }
 
-            reportInstallers.Add((builder, config) => config.WithReport(
-                new TraceReport(),
-                interval
-            ));
+        internal void SetUpObservers(ProbeContext probeContext)
+        {
+            registerObservers(probeContext);
         }
 
+        internal string ServiceControlMetricsAddress;
+        internal TimeSpan ReportingInterval;
+
+        Action<ProbeContext> registerObservers = c => {};
+    }
+
+    /// <summary>
+    /// Stores avialbe probes
+    /// </summary>
+    public class ProbeContext
+    {
+        /// <summary>
+        /// Duration type probes.
+        /// </summary>
+        public DurationProbe[] Durations;
+
         /// <summary>
-        /// Enables sending metric data to the NServiceBus log
+        /// Signal type probes.
         /// </summary>
-        /// <param name="interval">How often metric data is sent to the log</param>
-        /// <param name="logLevel">Level at which log entries should be written. Default is DEBUG.</param>
-        public void EnableLogTracing(TimeSpan interval, LogLevel logLevel = LogLevel.Debug)
+        public SignalProbe[] Signals;
```

Comment:
I've also changes properties to make sure that those cannot be changed when registering observers.